test/localization/check_duplicate_strings.js

```diff
@@ -0,0 +1,48 @@
+/*
+* Check that there are no duplicate strings in any individual l10n json file.
+ */
+var path = require('path');
+//var fs = require('fs');
+var tap = require('tap');
+
+var routes = require('../../src/routes.json');
+
+function noDuplicateValues (idsToCheck, name) {
+    var values = {};
+    for (var key in idsToCheck) {
+        if (values.hasOwnProperty(idsToCheck[key])) {
+            // duplicate values
+            //return false;
+            tap.fail(name + '.' + idsToCheck[key] + ' has duplicates');
+        } else {
+            values[idsToCheck[key]] = key;
+        }
+    }
+    tap.pass();
+    //return true;
+}
+
+tap.test('generalCheckForDuplicates', function (t) {
+    var ids = require(path.resolve(__dirname, '../../src/l10n.json'));
+    noDuplicateValues(ids, 'general');
+    t.end();
+});
+
+for (var v in routes) {
+    if (typeof routes[v].redirect !== 'undefined') {
+        continue;
+    }
+    var subdir = routes[v].view.split('/');
+    subdir.pop();
+    var name = routes[v].name;
+    try {
+        var ids = require(path.resolve(__dirname, '../../src/views/' + subdir.join('/') +'/l10n.json'));
```

Comment:
This would be a lot cleaner / more safe if done as:
```js
var fs = require('fs');
var path = require('path');

var uri = path.resolve(__dirname, './some/path/json.json');
var file = fs.readFileSync(uri, 'utf8');
if (file) {
    // do stuff after JSON.parse(file)
}
```